python-avd/scripts/custom_build_backend.py

```diff
@@ -58,26 +53,28 @@ def _insert_version() -> None:
             init_lines[index] = f'__version__ = "{version}"\n'
             break
 
-    with open(Path(__file__).parents[1].joinpath("pyavd/__init__.py"), mode="w", encoding="UTF-8") as init_file:
+    with Path(__file__).parents[1].joinpath("pyavd/__init__.py").open(mode="w", encoding="UTF-8") as init_file:
         init_file.writelines(init_lines)
 
 
-def get_requires_for_build_wheel(config_settings=None):
+def get_requires_for_build_wheel(config_settings: dict | None = None) -> list[str]:
     print("Fetch version from ansible.avd ansible collection and insert into __init__.py")
     _insert_version()
 
     print("Running 'make dep' to generate compiled Jinja2 templates and schemas pickle files.")
-    with Popen("make dep", shell=True) as make_process:
+    with Popen("make dep", shell=True) as make_process:  # noqa: S602,S607
```

Comment:
No I prefer to keep them in the code, since we then see them when we refactor, and avoid stale ignores in the pyproject.toml.
These two ignore stuff around shell and Popen. Not really important for these scripts.